### ISSUE TITLE
Regression supporting versions and dates in brackets and parenthsis

### ISSUE THREAD

**Issue by Blackjacx** [dblock/danger-changelog#37]
Hey there, 

yesterday I upgraded the plugin to version 0.4.0 and it started to produce the following errors:

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">One of the lines below found in CHANGELOG.md doesn’t match the expected format. Please make it look like the other lines, pay attention to version numbers, periods, spaces and date formats.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">There’re app changes, but not tests. That’s OK as long as you’re refactoring existing code.</td>
    </tr>
  </tbody>
</table>

```markdown
## [Unreleased]
```

```markdown
## [2.0.0] - 2018-12-11
```

```markdown
## [1.12.2] - 2018-11-26
```

```markdown
## [1.12.1] - 2018-11-26
```

```markdown
## [1.12.0] - 2018-11-21
```

```markdown
## [1.11.0] - 2018-11-09
```

```markdown
## [1.10.0] - 2018-11-05
```

```markdown
## [1.9.0] - 2018-10-26
```

```markdown
## [1.8.0] - 2018-10-11
```

```markdown
## [1.7.0] - 2018-10-05
```

```markdown
## [1.6.0] - 2018-09-18
```

```markdown
## [1.5.0] - 2018-08-31
```

```markdown
## [1.4.0] - 2018-08-20
```

```markdown
## [1.3.0] - 2018-08-03
```

```markdown
## [1.2.2] - 2018-08-03
```

```markdown
## [1.2.1] - 2018-07-24
```

```markdown
## [1.2.0] - 2018-07-20
```

```markdown
## [1.1.2] - 2018-07-13
```

```markdown
## [1.1.1] - 2018-06-28
```

```markdown
## [1.1.0] - 2018-06-27
```

```markdown
## [1.0.0] - 2018-06-08
```

```markdown
## [0.28.0] - 2018-06-06
```

```markdown
## [0.27.1] - 2018-05-28
```

```markdown
## [0.27.0] - 2018-05-25
```

```markdown
## [0.26.0] - 2018-04-28
```

```markdown
## [0.25.0] - 2018-03-28
```

```markdown
## [0.24.0] - 2018-03-20
```

```markdown
## [0.23.0] - 2018-03-05
```

```markdown
## [0.22.0] - 2018-02-19
```

```markdown
## [0.21.0] - 2018-02-02
```

```markdown
## [0.20.0] - 2018-01-20
```

```markdown
## [0.19.0] - 2018-01-05
```

```markdown
## [0.18.0] - 2017-12-15
```

```markdown
## [0.17.1] - 2017-11-28
```

```markdown
## [0.17.0] - 2017-11-17
```

```markdown
## [0.16.3] - 2017-11-14
```

```markdown
## [0.16.2] - 2017-11-07
```

```markdown
## [0.16.1] - 2017-11-03
```

```markdown
## [0.16.0] - 2017-10-23
```

```markdown
## [0.15.0] - 2017-10-23
```

```markdown
## [0.14.0] - 2017-10-13
```

```markdown
## [0.13.0] - 2017-10-09
```

```markdown
## [0.12.0] - 2017-09-20
```

```markdown
## [0.11.1] - 2017-09-15
```

```markdown
## [0.11.0] - 2017-09-14
```

```markdown
## [0.10.0] - 2017-09-08
```

```markdown
## [0.9.0] - 2017-08-31
```

```markdown
## [0.8.0] - 2017-08-26
```

```markdown
## [0.7.0] - 2017-08-21
```

```markdown
## [0.6.1] - 2017-08-18
```

```markdown
## [0.6.0] - 2017-08-14
```

```markdown
## [0.5.0] - 2017-08-04
```

```markdown
### Changed:
```

```markdown
### Fixed:
```

```markdown
## [0.4.0] - 2017-07-17
```

```markdown
## [0.3.1] - 2017-07-03
```

```markdown
## [0.3.0] - 2017-06-29
```

```markdown
## [0.2.2] - 2017-06-20
```

```markdown
## [0.2.1] - 2017-05-29
```

```markdown
## [0.2.0] - 2017-05-22
```

```markdown
## [0.1.1] - 2017-05-10
```

```markdown
## [0.1.0] - 2017-05-09
```

```markdown
## [0.0.3] - 2017-04-20
```

```markdown
## [0.0.2] - 2017-04-07
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

Help...

**Comment by dblock** [comment]
The format of dates got more strict. I have improved some things in https://github.com/dblock/danger-changelog/pull/38 but not all.

I think `[version]` is not going to be allowed, change it to just `version`.

The rest should be ok.

What do you think?

**Comment by Blackjacx** [comment]
> I think [version] is not going to be allowed, change it to just version.
Hmm why? This page still states that the braces are very valid and I like the syntax. Was there a need to change them? Now everybody needs to change their (huge) changelogs...

The date is actually auto generated by the fastlane-changelog plugin. It generates the release line like `## [2.0.0] - 2018-12-11`. In my opinion this is still a valid format. Important is consistency. Maybe it is a good idea to leave it up to the user which format is valid? If this stays as it is we apparently need to lock the your plugin to 0.3.0 :-(

**Comment by dblock** [comment]
Yeah no good reason other than not making things too complicated, but if this is some standard-ish or widely used format I can support it.

Btw, always accept PRs!

**Comment by Blackjacx** [comment]
Oh this would be awesome if you could support it 👍

**Comment by dblock** [comment]
I released 0.4.1 that supports all kinds of parenthesis and checks that they are balanced.

See #38 for details.

**Comment by dblock** [comment]
LMK if that works for you @Blackjacx!

**Comment by dblock** [comment]
Looks like `[Unreleased]` is still not working, I'll fix that.

**Comment by dblock** [comment]
Fixed in https://github.com/dblock/danger-changelog/commit/4702d7e1e3631f3a8555bc4c891ff75636919258

Care to try master on your CHANGELOG @Blackjacx? I'll cut a release when any remaining issues are resolved.

**Comment by Blackjacx** [comment]
Yeah with the 4.2 it works. It even found 2 inconsistencies in our changelog 👍

**Comment by dblock** [comment]
I released 0.4.2.

**Comment by Blackjacx** [comment]
Yeah this works! Thx 👍
